homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,215 @@
+"""Config flow for Plex."""
+import logging
+
+import plexapi.exceptions
+import requests.exceptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_URL,
+    CONF_HOST,
+    CONF_PORT,
+    CONF_TOKEN,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+
+from .const import (
+    CONF_SERVER,
+    CONF_SERVER_IDENTIFIER,
+    DEFAULT_PORT,
+    DEFAULT_SSL,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+    PLEX_SERVER_CONFIG,
+    SERVERS,
+)
+from .errors import NoServersFound, ServerNotSpecified
+from .server import PlexServer
+
+_LOGGER = logging.getLogger(__package__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class PlexFlowHandler(config_entries.ConfigFlow):
+    """Handle a Plex config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the Plex flow."""
+        self.current_login = {}
+        self.discovery_info = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            manual_setup = user_input.get("manual_setup")
```

Comment:
We should remove the manual step here. If the user wants a manual set up the user can use config yaml.

Reply:
This way has the benefits of:
1) Being pre-populated with the host:port of a local server if discovery is enabled.
2) Validating the config during runtime instead of requiring the user to restart with a new config to try again.

Reply:
We want config flows to be simple and streamlined. At least we shouldn't att this step in this PR.

Reply:
Okay, removed for now.